### DISCUSSION TITLE
Error while deploying fundme contract on sepolia

### DISCUSSION THREAD

**Discussion by johan784** [Cyfrin/foundry-full-course-cu#2891] [Q&A]
hey!

I am facing the following error while trying to deploy my contract on sepolia testnet

```
 forge script script/Deployfundme.s.sol:DeployFundMe \
    --rpc-url $SEPOLIA_RPC_URL \
    --private-key $SEPOLIA_PRIVATE_KEY \
    --broadcast \
    --verify \
    -vvvv

[⠊] Compiling...
No files changed, compilation skipped
Error: 
Could not find target contract
```

**Comment by LordAdhaar** [comment]
` forge script script/Deployfundme.s.sol:DeployFundMe \
    --rpc-url $SEPOLIA_RPC_URL \
    --private-key $SEPOLIA_PRIVATE_KEY \
    --broadcast \
    --verify \
    -vvvv
`

You might have named your contract wrong. Try,

` forge script script/Deployfundme.s.sol:Deployfundme --rpc-url $SEPOLIA_RPC_URL --private-key $SEPOLIA_PRIVATE_KEY --broadcast --verify -vvvv`

The stuff after ":" is the contract name you are targeting. In you github repo your contract is called "Deployfundme" but here you are using "DeployFundMe". That is probably causing the issue. Let me know if this works.

**Comment by EngrPips** [comment]
Great heads-up

**Comment by chainflash** [comment]
remove this part and try again : **:DeployFundMe**
or copy the command I wrote below and paste it in the terminal :
```
forge script script/DeploySimpleStorage.s.sol --rpc-url $SEPOLIA_RPC_URL --private-key $PRIVATE_KEY --broadcast
```
Of course, it is better not to put the private key in plain text 👍

**Reply by johan784** [reply]
tried hardcoding the private key as well but my terminal says I need a fork url although I have provided it with one :(

**Comment by 0xivanalexander** [comment]
Sometimes I get the same error and I just run forge clean forge build and everything starts to work as expected

**Reply by EngrPips** [reply]
Interesting

**Comment by LordAdhaar** [comment]
Bro left us in suspense

**Reply by johan784** [reply]
hey! I did try executing the command with the changes you mentioned but ive been receiving another error which seems to be happening often to me, here's the error:


```
error: a value is required for '--fork-url <URL>' but none was supplied
```

I have provided a fork url and made sure my .env file contains all the necessary keys  and RPC url 
yet it asks me to provide a fork url

**Reply by EngrPips** [reply]
If you run `source .env` in your terminal and then run `echo $SEPOLIA_RPC_URL` what do you get as output?

**Reply by johan784** [reply]
I get the alchemy url I've obtained for sepolia

**Reply by EngrPips** [reply]
`forge script script/DeployFundMe.s.sol:DeployFundMe --rpc-url $SEPOLIA_RPC_URL --private-key $SEPOLIA_PRIVATE_KEY --broadcast --verify --etherscan-api-key $ETHERSCAN_API_KEY -vvvv`

If you run the above command and have the folders and file names matching what is in the above command, then the command is supposed to work as expected.

![Screenshot 2024-11-22 at 08 51 50](https://github.com/user-attachments/assets/a2846694-7ea0-4fbd-bdde-b5128052a604)

**Reply by johan784** [reply]
alright, thanks!

**Reply by EngrPips** [reply]
Anytime.

**Comment by EngrPips** [comment]
It is what it is.

**Comment by plamenkoyovchev** [comment]
Hello guys, I'm also getting an error while trying to use fork url:

```
forge test --mt testPriceFeedVersionIsAccurate --fork-url $SEPOLIA_RPC_URL
[⠊] Compiling...
No files changed, compilation skipped
Error: Could not instantiate forked environment with provider eth-sepolia.g.alchemy.com

Context:
- Error #0: Failed to get latest block number
- Error #1: HTTP error 401 with body: {"jsonrpc":"2.0","id":0,"error":{"code":-32600,"message":"Must be authenticated!"}}
- Error #2: HTTP error 401 with body: {"jsonrpc":"2.0","id":0,"error":{"code":-32600,"message":"Must be authenticated!"}}
```

**Reply by EngrPips** [reply]
I think your `api-key` is incorrect.

**Reply by plamenkoyovchev** [reply]
Correct, I've forgotten to run `source .env` when I updated my api key.
